recipes/openjph/all/patches/0.15.0_cmake-cxx-standard.patch

```diff
@@ -0,0 +1,26 @@
+diff --git CMakeLists.txt CMakeLists.txt
+index cfa70ea..7e761db 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -93,7 +93,7 @@ message(STATUS "Building ${CMAKE_BUILD_TYPE}")
+ 
+ ## C++ version and flags
+ # C++14 is needed for gtest, otherwise, C++11 is sufficient for the library
+-set(CMAKE_CXX_STANDARD 14)
++#set(CMAKE_CXX_STANDARD 14)
+ if (MSVC)
+   add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+ endif()
+diff --git src/apps/ojph_stream_expand/CMakeLists.txt src/apps/ojph_stream_expand/CMakeLists.txt
+index 5aea342..e21bdd9 100644
+--- src/apps/ojph_stream_expand/CMakeLists.txt
++++ src/apps/ojph_stream_expand/CMakeLists.txt
+@@ -5,7 +5,7 @@ if (OJPH_BUILD_STREAM_EXPAND)
+ 
+   include_directories(../common)
+   include_directories(../../core/common)
+-  set(CMAKE_CXX_STANDARD 14)
++  #set(CMAKE_CXX_STANDARD 14)
```

Comment:
@AbrilRBS, @irieger 

The issue in Conan 1.x on macOS is happening because the Conan 1 profiles do not define `compiler.cppstd`, and the `appel-clang` compiler defaults to `gnu98` on the command line.
Fully commenting out this line will cause these Conan 1 builds to _not_ pass any standard whatsoever, and default to whatever the compiler defaults to - clearly something 11 or greter on all others except for apple clang.

The recipe probably needs something like
``` 
if not self.settings.get_safe("compiler.cppstd"):
    tc.cache_variables["CMAKE_CXX_STANDARD"] = xx # 11 or 14 depending on some condition
```

alternatively maybe tweaking the patch to make it a cache variable _might_ work :
* default to that value in Conan 1.x when cppstd is not defined
* obey the value in the `conan_toolchain.cmake` if provided, ignoring the hardcoded value in the cmakelists

but this latter alternative would need to be tested

Reply:
Ok, your first suggestion did work.

Reply:
Thank you!